src/main/java/life/qbic/expdesign/io/MHCLigandDesignReader.java

```diff
@@ -138,7 +139,32 @@ public MHCLigandDesignReader() {
     headersToTypeCodePerSampletype.put(SampleType.Q_MHC_LIGAND_EXTRACT, ligandsMetadata);
     // headersToTypeCodePerSampletype.put("Q_MS_RUN", msRunMetadata);
   }
+  private void fillParsedCategoriesToValuesForRow(Map<String, Integer> headerMapping,
+      String[] row) {
+    // logger.info("Collecting possible CV entries for row.");
+    addValueForCategory(headerMapping, row, "Organism");
+    addValueForCategory(headerMapping, row, "Tissue");
+    addValueForCategory(headerMapping, row, "Dignity");
+    addValueForCategory(headerMapping, row, "Tumor Type");
+    addValueForCategory(headerMapping, row, "TNM");
+    addValueForCategory(headerMapping, row, "Antibody");
+    addValueForCategory(headerMapping, row, "MHC Class");
+  }
 
+  private void addValueForCategory(Map<String, Integer> headerMapping, String[] row, String cat) {
+    if (headerMapping.containsKey(cat)) {
+      String val = row[headerMapping.get(cat)];
+      if (val != null && !val.isEmpty()) {
```

Comment:
NIT: Should the outer if cases be logged? Alternatively you could think about introducing optional's here to reduce the need for nested if cases?